Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -130,8 +130,66 @@ open class BufferingBookmarksSynchronizer: TimestampedSingleCollectionSynchroniz
                         return deferMaybe(result)
                     }
                 }
-
                 return deferMaybe(result)
+            } >>== { result in
+                guard case .completed = result else {
```

Comment:
You also need to handle the case of no new data; if you short-circuit here, clients won't upload unless _another device_ has added data.